libs/application/templates/announcement-of-death/src/lib/dataSchema.ts

```diff
@@ -37,52 +36,41 @@ export const dataSchema = z.object({
       }),
     })
     .partial()
+    .refine(({ roleConfirmation }) => !!roleConfirmation, {
+      message: m.errorRoleConfirmation.defaultMessage,
+      path: ['roleConfirmation'],
+    })
     .refine(
       ({ roleConfirmation, electPerson }) =>
-        (roleConfirmation === RoleConfirmationEnum.DELEGATE &&
-        electPerson?.electedPersonName
-          ? electPerson?.electedPersonName !== ''
-          : false) ||
-        (roleConfirmation === RoleConfirmationEnum.CONTINUE &&
-          (electPerson?.electedPersonName === '' ||
-            electPerson?.electedPersonName !== '')),
+        roleConfirmation === RoleConfirmationEnum.DELEGATE && !!electPerson
+          ? electPerson.electedPersonName !== '' &&
+            electPerson.electedPersonNationalId !== ''
+          : !electPerson
+          ? true
+          : roleConfirmation === RoleConfirmationEnum.CONTINUE
+          ? true
+          : false,
       {
         message: m.errorNationalIdNoName.defaultMessage,
         path: ['electPerson', 'electedPersonNationalId'],
       },
     )
     .refine(
       ({ roleConfirmation, electPerson }) =>
-        (roleConfirmation === RoleConfirmationEnum.DELEGATE &&
-        electPerson?.electedPersonNationalId
-          ? kennitala.isPerson(electPerson?.electedPersonNationalId)
-          : false) ||
-        (roleConfirmation === RoleConfirmationEnum.CONTINUE &&
-          (electPerson?.electedPersonNationalId === '' ||
-            electPerson?.electedPersonNationalId !== '')),
+        roleConfirmation === RoleConfirmationEnum.DELEGATE && !!electPerson
+          ? electPerson.electedPersonNationalId ??
```

Comment:
```suggestion
          ? electPerson.electedPersonNationalId &&
```
The national id is evaluated as truthy and therefore everything behind ?? won't be evaluated?